packages/clerk-js/src/ui/hooks/usePasswordComplexity.ts

```diff
@@ -0,0 +1,126 @@
+import type { PasswordSettingsData } from '@clerk/types';
+import { useCallback, useMemo, useState } from 'react';
+
+import { localizationKeys, useLocalizations } from '../localization';
+
+const testComplexityCases = (
+  password: string,
+  {
+    minLength,
+    maxLength,
+  }: {
+    minLength: number;
+    maxLength: number;
+  },
+) => {
+  return {
+    max_length: password.length < maxLength,
+    min_length: password.length >= minLength,
+    require_numbers: /\d/.test(password),
+    require_lowercase: /[a-z]/.test(password),
+    require_uppercase: /[A-Z]/.test(password),
+    require_special_char: /[!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~]/.test(password),
+  };
+};
+
+type ComplexityErrorMessages = {
```

Comment:
Is this equivalent? 
```suggestion
type ComplexityErrorMessages = Partial<Record<Omit<PasswordSettingsData, 'disable_hibp' | 'min_zxcvbn_strength' | 'show_zxcvbn'>>, string>
```

Reply:
Seems like it but then I have to handle the following TS error

```
Type 'Omit ' does not satisfy the constraint 'string | number | symbol'.
```